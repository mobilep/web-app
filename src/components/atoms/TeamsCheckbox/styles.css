.TeamsCheckbox {
	display: flex;
	align-items: center;
	position: relative;
}

.TeamsCheckbox input {
	top: 0;
	left: 0;
	position: absolute;
	opacity: 0;
	cursor: pointer;
	height: 22px;
	width: 22px;
}

.TeamsCheckbox-checkmark {
	height: 22px;
	width: 22px;
	border-radius: 50%;
	color: transparent;
	border: 1px solid var(--almostBlack);
	margin-right: 20px;
	padding-right: 20px;
}

.TeamsCheckbox-checkmark .svgWrapper {
	max-width: 24px;
}

.TeamsCheckbox input:checked ~ .TeamsCheckbox-checkmark {
	background-color: var(--tomatoRed);
	color: white;
	border: none;
}

.Teamscheckbox-name {
	margin-left: 10px;
}
