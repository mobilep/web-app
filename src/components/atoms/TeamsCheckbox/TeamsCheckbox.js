import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { Avatar } from '..';
import constants from '../Avatar';
import { CheckIcon } from '../../../icons';

import './styles.css';

export default class TeamsCheckbox extends Component {

	static propTypes = {
		avatar: PropTypes.string,
		avatarBgColor: PropTypes.string,
		avatarInitials: PropTypes.string,
		checked: PropTypes.bool,
		className: PropTypes.string,
		editMode: PropTypes.bool,
		id: PropTypes.string,
		name: PropTypes.string,
		onInputChange: PropTypes.func,
	}

	getClassName = (className) => {
		return classnames({
			TeamsCheckbox: true,
		}, className);
	}

	render () {
		const {
			avatar,
			avatarBgColor,
			avatarInitials,
			checked,
			className,
			editMode,
			id,
			name,
			onInputChange,
		} = this.props;

		const { matches } = window.matchMedia('(max-width: 768px)');

		return (
			<label className={this.getClassName(className)}>
				{editMode
					? <Fragment>
						<input type='checkbox'
							name={id}
							onChange={onInputChange}
							defaultChecked={checked}
						/>
						<span className='TeamsCheckbox-checkmark'>
							<CheckIcon />
						</span>
					</Fragment>
					: null}
				<Avatar
					initials={avatarInitials}
					img={avatar}
					size={!matches ? constants.AVATAR_SIZES.AVATAR_60 : constants.AVATAR_SIZES.AVATAR_50}
					border={false}
					color={avatarBgColor}
				/>
				<span className='Teamscheckbox-name'>{name}</span>
			</label>
		);
	}
}
