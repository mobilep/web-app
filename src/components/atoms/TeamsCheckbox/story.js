import React from 'react';
import { storiesOf } from '@storybook/react';
import { boolean } from '@storybook/addon-knobs';
import TeamsCheckbox from './TeamsCheckbox';

storiesOf('1. Atoms / TeamsCheckbox', module)
	.add('default', () =>
		(<TeamsCheckbox
			avatarInitials='NN'
			avatar='https://www.elastic.co/assets/bltada7771f270d08f6/enhanced-buzz-1492-1379411828-15.jpg'
			avatarBgColor='#F3423C'
			name='Myra Holt'
			editMode={boolean('Edit mode', false)}
		/>));
