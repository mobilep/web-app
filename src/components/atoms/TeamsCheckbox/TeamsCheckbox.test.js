import React from 'react';
import { shallow } from 'enzyme';
import mockMatchMedia from '../../../helpers/test/matchMedia';

import TeamsCheckbox from './TeamsCheckbox';

describe('Atom - TeamsCheckbox', () => {

	beforeAll(() => mockMatchMedia());

	it('Should render default mode', () => {
		const wrapper = shallow(
			<TeamsCheckbox
				avatarInitials='NN'
				avatar='https://www.elastic.co/assets/bltada7771f270d08f6/enhanced-buzz-1492-1379411828-15.jpg'
				avatarBgColor='#F3423C'
				name='Myra Holt'
			/>);
		expect(wrapper).toMatchSnapshot();
	});

	it('Should render edit mode', () => {
		const wrapper = shallow(
			<TeamsCheckbox
				avatarInitials='NN'
				avatar='https://www.elastic.co/assets/bltada7771f270d08f6/enhanced-buzz-1492-1379411828-15.jpg'
				avatarBgColor='#F3423C'
				name='Myra Holt'
				editMode
			/>);
		expect(wrapper).toMatchSnapshot();
	});
});
