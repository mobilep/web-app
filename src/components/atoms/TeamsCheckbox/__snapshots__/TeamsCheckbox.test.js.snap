// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom - TeamsCheckbox Should render default mode 1`] = `
<label
  className="TeamsCheckbox"
>
  <Avatar
    border={false}
    color="#F3423C"
    img="https://www.elastic.co/assets/bltada7771f270d08f6/enhanced-buzz-1492-1379411828-15.jpg"
    initials="NN"
    size="avatar-50"
  />
  <span
    className="Teamscheckbox-name"
  >
    Myra Holt
  </span>
</label>
`;

exports[`Atom - TeamsCheckbox Should render edit mode 1`] = `
<label
  className="TeamsCheckbox"
>
  <input
    type="checkbox"
  />
  <span
    className="TeamsCheckbox-checkmark"
  >
    <CheckIcon />
  </span>
  <Avatar
    border={false}
    color="#F3423C"
    img="https://www.elastic.co/assets/bltada7771f270d08f6/enhanced-buzz-1492-1379411828-15.jpg"
    initials="NN"
    size="avatar-50"
  />
  <span
    className="Teamscheckbox-name"
  >
    Myra Holt
  </span>
</label>
`;
